relayer/chains/cosmos/feegrant.go

```diff
@@ -299,72 +340,58 @@ func (cc *CosmosProvider) EnsureBasicGrants(ctx context.Context, memo string) (*
 
 		granterExists := cc.EnsureExists(cliCtx, granterAcc) == nil
 
-		//Feegranter exists on chain
+		// Feegranter exists on chain
 		if granterExists {
-			txResp, err := cc.SubmitTxAwaitResponse(ctx, msgs, memo, 0, granterKey)
+			txResp, err := cc.SubmitTxAwaitResponse(ctx, msgs, memo, gas, granterKey)
 			if err != nil {
-				fmt.Printf("Error: SubmitTxAwaitResponse: %s", err.Error())
 				return nil, err
 			} else if txResp != nil && txResp.TxResponse != nil && txResp.TxResponse.Code != 0 {
-				fmt.Printf("Submitting grants for granter %s failed. Code: %d, TX hash: %s\n", granterKey, txResp.TxResponse.Code, txResp.TxResponse.TxHash)
+				cc.log.Warn("Feegrant TX failed", zap.String("TX hash", txResp.TxResponse.TxHash), zap.Uint32("code", txResp.TxResponse.Code))
 				return nil, fmt.Errorf("could not configure feegrant for granter %s", granterKey)
 			}
 
-			fmt.Printf("TX succeeded, %d new grants configured, %d grants already in place. TX hash: %s\n", grantsNeeded, numGrantees-grantsNeeded, txResp.TxResponse.TxHash)
+			cc.log.Info("Feegrant succeeded", zap.Int("new grants", grantsNeeded), zap.Int("existing grants", numGrantees-grantsNeeded), zap.String("TX hash", txResp.TxResponse.TxHash))
 			return txResp.TxResponse, err
-		} else {
-			return nil, fmt.Errorf("granter %s does not exist on chain", granterKey)
 		}
-	} else {
-		fmt.Printf("All grantees (%d total) already had valid feegrants. Feegrant configuration verified.\n", numGrantees)
-	}
 
-	return nil, nil
-}
-
-func getGasTokenDenom(gasPrices string) (string, error) {
-	r := regexp.MustCompile(`(?P<digits>[0-9.]*)(?P<denom>.*)`)
-	submatches := r.FindStringSubmatch(gasPrices)
-	if len(submatches) != 3 {
-		return "", errors.New("could not find fee denom")
+		return nil, fmt.Errorf("granter %s does not exist on chain", granterKey)
 	}
 
-	return submatches[2], nil
+	return nil, nil
 }
 
 // GrantBasicAllowance Send a feegrant with the basic allowance type.
 // This function does not check for existing feegrant authorizations.
 // TODO: check for existing authorizations prior to attempting new one.
 func (cc *CosmosProvider) GrantAllGranteesBasicAllowance(ctx context.Context, gas uint64) error {
 	if cc.PCfg.FeeGrants == nil {
-		return errors.New("ChainClient must be a FeeGranter to establish grants")
+		return errors.New("chain client must be a FeeGranter to establish grants")
 	} else if len(cc.PCfg.FeeGrants.ManagedGrantees) == 0 {
-		return errors.New("ChainClient is a FeeGranter, but is not managing any Grantees")
+		return errors.New("chain client is a FeeGranter, but is not managing any Grantees")
 	}
 
-	granterKey := cc.PCfg.FeeGrants.GranterKey
+	granterKey := cc.PCfg.FeeGrants.GranterKeyOrAddr
 	if granterKey == "" {
 		granterKey = cc.PCfg.Key
 	}
 	granterAddr, err := cc.GetKeyAddressForKey(granterKey)
 	if err != nil {
-		fmt.Printf("ChainClient FeeGranter misconfiguration: %s", err.Error())
+		cc.log.Error("Unknown granter", zap.String("key name", granterKey))
```

Comment:
nit: log field names should be snake_case https://github.com/cosmos/relayer/blob/main/docs/logging.md#usage